src/main/java/org/myrobotlab/service/ServoMixer.java

```diff
@@ -175,305 +256,345 @@ public Pose loadPose(String name) {
    * @return the loaded pose object
    */
   public Pose getPose(String name) {
-    Pose pose = null;
+    
     try {
-      if (!checkDir(servoMixerDirectory)) {
-        error("invalid poses directory %s", servoMixerDirectory);
-        return null;
-      }
-      String filename = new File(servoMixerDirectory).getAbsolutePath() + File.separator + name + ".pose";
-      log.info("Loading Pose name {}", filename);
-      pose = Pose.loadPose(filename);
+      ServoMixerConfig c = (ServoMixerConfig)config;
+
+      String filename = new File(c.posesDir).getAbsolutePath() + File.separator + name + ".yml";
+      log.info("loading pose name {}", filename);
+      String yml = FileIO.toString(filename);
+      return CodecUtils.fromYaml(yml, Pose.class);
+      // pose = Pose.loadPose(filename);
       // broadcastState(); "maybe too chatty"
     } catch (Exception e) {
       error(e);
     }
-    return pose;
+    return null;
   }
 
-  public void removePose(String name) {
-    try {
-      if (!checkDir(servoMixerDirectory)) {
-        error("invalid poses directory %s", servoMixerDirectory);
-        return;
+  public List<String> getPoseFiles() {
+    ServoMixerConfig c = (ServoMixerConfig)config;
+
+    List<String> files = new ArrayList<>();
+    if (!FileIO.checkDir(c.posesDir)) {
+      return files;
+    }
+
+    File dir = new File(c.posesDir);
+
+    if (!dir.exists() || !dir.isDirectory()) {
+      error("%s not a valid directory", c.posesDir);
+      return files;
+    }
+    File[] all = dir.listFiles();
+    Set<String> sorted = new TreeSet<>();
+    for (File f : all) {
+      if (f.getName().toLowerCase().endsWith(".yml")) {
+        sorted.add(f.getName().substring(0, f.getName().lastIndexOf(".")));
       }
-      String filename = new File(servoMixerDirectory).getAbsolutePath() + File.separator + name + ".pose";
-      File del = new File(filename);
-      if (del.exists()) {
-        del.delete();
+    }
+    for (String s : sorted) {
+      files.add(s);
+    }
+    return files;
+  }
+
+  public String getPosesDirectory() {
+    ServoMixerConfig c = (ServoMixerConfig)config;
+    return c.posesDir;
+  }
+
+  public List<ServoControl> listAllServos() {

```

Comment:
Shouldn't be needed, equivalent to `Runtime.getServicesFromInterface(ServoControl.class)`. Speaking of which, that method is terribly unoptimized and could use some cleaning up

Reply:
Service.interfaceSet  is the way to go !   The webguis use this, but java-land could be using this more too...
We should:
* change it to a TreeSet not Map
* have it done only "once" then services of the same type just need a reference to it